cli/src/cmd/forge/script/mod.rs

```diff
@@ -451,10 +453,47 @@ impl ScriptArgs {
     /// the user.
     fn check_contract_sizes(
         &self,
-        transactions: Option<&VecDeque<TypedTransaction>>,
+        result: &ScriptResult,
         known_contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
     ) -> eyre::Result<()> {
-        for (data, to) in transactions.iter().flat_map(|txes| {
+        // (name, init, deployed)[]
+        let mut bytecodes: Vec<(String, &[u8], &[u8])> = vec![];
+
+        // From artifacts
+        known_contracts.iter().for_each(|(artifact, bytecode)| {
+            const MSG: &str = "Code should have been linked before.";
+            let init_code = bytecode.bytecode.object.as_bytes().expect(MSG);
+            // Ignore abstract contracts
+            if let Some(ref deployed_code) = bytecode.deployed_bytecode.bytecode {
+                let deployed_code = deployed_code.object.as_bytes().expect(MSG);
+                bytecodes.push((artifact.name.clone(), init_code, deployed_code));
+            }
+        });
+
+        // From traces, only push if it was not present already
+        let create_nodes = result.traces.iter().flat_map(|(_, traces)| {
+            traces
+                .arena
+                .iter()
+                .filter(|node| matches!(node.kind(), CallKind::Create | CallKind::Create2))
+        });
+
+        let mut i = 0usize;
+        for node in create_nodes {
```

Comment:
Why do we need the index here?

Reply:
would be to differentiate multiple unnamed contracts, should be in the same order that the traces are displayed